de.metas.serviceprovider/src/main/java/de/metas/serviceprovider/external/project/ExternalProjectRepository.java

```diff
@@ -39,21 +38,41 @@
 @Repository
 public class ExternalProjectRepository
 {
-	private final IQueryBL queryBL = Services.get(IQueryBL.class);
+	private final IQueryBL queryBL;
+
+	public ExternalProjectRepository(final IQueryBL queryBL)
+	{
+		this.queryBL = queryBL;
+	}
 
 	@NonNull
 	public ImmutableList<ExternalProjectReference> getByExternalSystem(@NonNull final ExternalSystem externalSystem)
 	{
 		return queryBL.createQueryBuilder(I_S_ExternalProjectReference.class)
 				.addOnlyActiveRecordsFilter()
 				.addEqualsFilter(I_S_ExternalProjectReference.COLUMN_ExternalSystem, externalSystem.getValue())
+				.orderBy(I_S_ExternalProjectReference.COLUMNNAME_SeqNo)
 				.create()
 				.list()
 				.stream()
 				.map(this::buildExternalProjectReference)
 				.collect(ImmutableList.toImmutableList());
 	}
 
+	@NonNull
+	public Optional<ExternalProjectReference> getByRequestOptional(@NonNull final GetExternalProjectRequest getExternalProjectRequest)
+	{
+		return queryBL.createQueryBuilder(I_S_ExternalProjectReference.class)
+				.addEqualsFilter(I_S_ExternalProjectReference.COLUMN_ExternalSystem, getExternalProjectRequest.getExternalSystem().getValue())
```

Comment:
don't forget to add `.addOnlyActiveRecordsFilter()`, as a general rule of thumb.